exec/java-exec/src/test/java/org/apache/drill/exec/rpc/user/security/TestUserBitKerberos.java

```diff
@@ -137,6 +142,41 @@ public Void run() throws Exception {
     test("SELECT * FROM cp.`region.json` LIMIT 5");
   }
 
+  @Test
+  public void testUnecryptedConnectionCounter() throws Exception {
+    final Properties connectionProps = new Properties();
+    connectionProps.setProperty(DrillProperties.SERVICE_PRINCIPAL, krbHelper.SERVER_PRINCIPAL);
+    connectionProps.setProperty(DrillProperties.KERBEROS_FROM_SUBJECT, "true");
+    final Subject clientSubject = JaasKrbUtil.loginUsingKeytab(krbHelper.CLIENT_PRINCIPAL,
+        krbHelper.clientKeytab.getAbsoluteFile());
+
+    Subject.doAs(clientSubject, new PrivilegedExceptionAction<Void>() {
+      @Override
+      public Void run() throws Exception {
+        updateClient(connectionProps);
+        return null;
+      }
+    });
+
+    // Run few queries using the new client
+    testBuilder()
+        .sqlQuery("SELECT session_user FROM (SELECT * FROM sys.drillbits LIMIT 1)")
+        .unOrdered()
+        .baselineColumns("session_user")
+        .baselineValues(krbHelper.CLIENT_SHORT_NAME)
+        .go();
+
+    // Check encrypted counters value
```

Comment:
Updated the test description for explaining the logic behind different count values. Also rebase PR on latest apache master. This test will change based on the [PR for DRILL-5721](https://github.com/apache/drill/pull/919) since for local fragment's on Foreman node status update will happen locally and hence there won't be any Control Connection created from a Drillbit to itself. I will update the test based on which commit makes in first.